src/Microsoft.ApplicationInsights.AspNet/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -53,41 +54,16 @@ public static void AddApplicationInsightsTelemetry(this IServiceCollection servi
                 return telemetryConfiguration;
             });
 
+            services.AddSingleton<IJavaScriptSnippet, ApplicationInsightsJavaScript>();
+
             services.AddScoped<TelemetryClient>();
 
             services.AddScoped<RequestTelemetry>((svcs) => {
```

Comment:
Ah, yes. I remember that quirk. Might be a good idea to leave a comment and prevent someone from "simplifying" it by accident :smile: